internal/services/network/nat_gateway_public_ip_v6_association_resource.go

```diff
@@ -0,0 +1,260 @@
+				}
+			}
+
+			publicIpAddressesV6 = append(publicIpAddressesV6, natgateways.SubResource{
+				Id: pointer.To(state.PublicIpAddressId),
+			})
+			natGateway.Model.Properties.PublicIPAddressesV6 = pointer.To(publicIpAddressesV6)
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *natGatewayId, *natGateway.Model); err != nil {
+				return fmt.Errorf("updating %s: %+v", natGatewayId, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+		Timeout: 30 * time.Minute,
+	}
+}
+
+func (r NatGatewayPublicIpV6AssociationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.NatGateways
+
+			id, err := commonids.ParseCompositeResourceID(metadata.ResourceData.Id(), &natgateways.NatGatewayId{}, &commonids.PublicIPAddressId{})
+			if err != nil {
+				return err
+			}
+
+			natGateway, err := client.Get(ctx, *id.First, natgateways.DefaultGetOperationOptions())
+			if err != nil {
+				if response.WasNotFound(natGateway.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id.First, err)
+			}
+
+			if natGateway.Model == nil || natGateway.Model.Properties == nil {
+				return metadata.MarkAsGone(id)
+			}
+
+			if natGateway.Model.Properties.PublicIPAddressesV6 == nil {
+				return metadata.MarkAsGone(id)
+			}
+
+			publicIPAddressFound := false
+			for _, pip := range *natGateway.Model.Properties.PublicIPAddressesV6 {
+				if pip.Id == nil {
+					continue
+				}
+
+				if strings.EqualFold(*pip.Id, id.Second.ID()) {
+					publicIPAddressFound = true
+					break
+				}
+			}
+
+			if !publicIPAddressFound {
+				return metadata.MarkAsGone(id)
+			}
+
+			state := NatGatewayPublicIpV6AssociationModel{
+				NatGatewayId:      id.First.ID(),
+				PublicIpAddressId: id.Second.ID(),
+			}
+
+			return metadata.Encode(&state)
+		},
+		Timeout: 5 * time.Minute,
+	}
+}
+
+func (r NatGatewayPublicIpV6AssociationResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.NatGateways
+
+			var state NatGatewayPublicIpV6AssociationModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			publicIpAddressId, err := commonids.ParsePublicIPAddressID(state.PublicIpAddressId)
+			if err != nil {
+				return err
+			}
+
+			natGatewayId, err := natgateways.ParseNatGatewayID(state.NatGatewayId)
+			if err != nil {
+				return err
+			}
+
+			locks.ByName(natGatewayId.NatGatewayName, natGatewayResourceName)
+			defer locks.UnlockByName(natGatewayId.NatGatewayName, natGatewayResourceName)
+
+			natGateway, err := client.Get(ctx, *natGatewayId, natgateways.DefaultGetOperationOptions())
+			if err != nil {
+				if response.WasNotFound(natGateway.HttpResponse) {
+					return fmt.Errorf("%s was not found", natGatewayId)
```

Comment:
Pointers used in `fmt.Errorf()` should be dereferenced, please fix similar issues in the _resource file.
```suggestion
					return fmt.Errorf("%s was not found", *natGatewayId)
```

Reply:
Updated